File: app/src/main/java/com/example/mobv2/callbacks/GetCommentCallback.java
package com.example.mobv2.callbacks;

import android.util.Log;

import androidx.recyclerview.widget.RecyclerView;

import com.example.mobv2.adapters.CommentsAdapter;
import com.example.mobv2.models.Comment;
import com.example.mobv2.serverapi.MOBServerAPI;
import com.example.mobv2.ui.activities.MainActivity;
import com.google.gson.internal.LinkedTreeMap;

public class GetCommentCallback implements MOBServerAPI.MOBAPICallback
{
    protected final MainActivity mainActivity;
    protected final RecyclerView commentsRecycler;

    public GetCommentCallback(MainActivity mainActivity, RecyclerView commentsRecycler)
    {
        this.mainActivity = mainActivity;
        this.commentsRecycler = commentsRecycler;
    }

    @Override
    public void funcOk(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());

        var response = (LinkedTreeMap<String, Object>) obj.get("response");

        Comment comment = new Comment.CommentBuilder().parseFromMap(response);
        var commentsAdapter = (CommentsAdapter) commentsRecycler.getAdapter();
        commentsAdapter.addComment(comment);
        commentsRecycler.scrollToPosition(0);
    }

    @Override
    public void funcBad(LinkedTreeMap<String, Object> obj)
    {
        Log.v("DEBUG", obj.toString());
    }

    @Override
    public void fail(Throwable obj)
    {
        Log.v("DEBUG", obj.toString());
    }
}

File: app/src/main/java/com/example/mobv2/callbacks/GetCommentCallback.kt
package com.example.mobv2.callbacks

import android.util.Log
import com.example.mobv2.callbacks.abstractions.GetCommentFailCallback
import com.example.mobv2.callbacks.abstractions.GetCommentOkCallback
import com.example.mobv2.ui.activities.MainActivity
import com.example.mobv2.utils.abstractions.responseStateCallbacks.HasFailStateCallback
import com.example.mobv2.utils.abstractions.responseStateCallbacks.HasOkStateCallback
import com.google.gson.internal.LinkedTreeMap
import serverapi.MOBServerAPI.MOBAPICallback

class GetCommentCallback(private val mainActivity: MainActivity) : MOBAPICallback,
    HasOkStateCallback<GetCommentOkCallback>, HasFailStateCallback<GetCommentFailCallback>
{
    private var okCallback: GetCommentOkCallback? = null
    private var failCallback: GetCommentFailCallback? = null
    override fun funcOk(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
        val response = obj["response"] as LinkedTreeMap<String, Any>?

        okCallback?.parseCommentFromMapAndAddToComments(response)
    }

    override fun funcBad(obj: LinkedTreeMap<String, Any>)
    {
        Log.v("DEBUG", obj.toString())
    }

    override fun fail(obj: Throwable)
    {
        Log.v("DEBUG", obj.toString())

        failCallback?.onDisconnect()
    }

    override fun setOkCallback(okCallback: GetCommentOkCallback)
    {
        this.okCallback = okCallback
    }

    override fun setFailCallback(failCallback: GetCommentFailCallback)
    {
        this.failCallback = failCallback
    }
}